pkg/storage/replica_state.go

```diff
@@ -210,6 +210,47 @@ func setAppliedIndex(
 		nil /* txn */)
 }
 
+func setAppliedIndexBlind(
```

Comment:
Yes. We use the non-blind version during in `saveState` where I'm not 100% sure that the previous state is empty.